package.json

```diff
@@ -143,6 +144,7 @@
     "numeral": "^2.0.6",
     "react-app-rewire-styled-components": "^3.0.2",
     "react-app-rewired": "^2.1.8",
+    "react-copy-to-clipboard": "^5.0.4",
```

Comment:
I think we dont need to use an external lib here. We can use a helper function like this:

```
export function copyToClipboard(textToCopy: string) {
  // navigator clipboard api needs a secure context (https)
  if (navigator.clipboard && window.isSecureContext) {
    // navigator clipboard api method'
    return navigator.clipboard.writeText(textToCopy)
  } else {
    // text area method
    const textArea = document.createElement('textarea')
    textArea.value = textToCopy
    // make the textarea out of viewport
    textArea.style.position = 'fixed'
    textArea.style.left = '-999999px'
    textArea.style.top = '-999999px'
    document.body.appendChild(textArea)
    textArea.focus()
    textArea.select()
    return new Promise((res, rej) => {
      // here the magic happens
      document.execCommand('copy') ? res() : rej()
      textArea.remove()
    })
  }
}
```

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand
base on this link, document.execCommand will be deprecated so I think should not use it. Also, I had a bug in Android Metamask's browser that was tricky to find a workaround. So I think to support for any browser/device we should use a library and also its will update in future if something got deprecated.

Reply:
yeah, I know that issue which you face, some browser do not support `navigator` yet, so our site will be crashed, I faced that issue before. That function I wrote above also is a workaround. And in fact, if browser do not support `document.execCommand`, I think that browser will support `navigator`, and vice versa, if browser do not support navigator, It will support `document.execCommand`

Related to document.execCommand will be deprecated, Actually, the library which you use are using document.execCommand too :D. Refer here: https://github.com/sudodoki/copy-to-clipboard/blob/master/index.js#L77 (react-copy-to-clipboard use copy-to-clipboard lib)

Reply:
I knew they use the document.execCommand too, I have read their code, but as I said, it will deprecate, and the lib will update if have any issues. 
And in this task, I need to handle after-copy logic, I can do it manually but the code looks ugly, hard to read,...
Also, this lib is very small size and has a very high reputation. So I don't see why we can't use it for better maintainable

Reply:
but anyway, your decision about this.